src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -672,36 +678,80 @@ private static string GetCommonSuffix(string firstPath, string secondPath)
         private void AttachToDocumentChanges(uint docCookie, IVsWindowFrame pFrame)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (RunIndexToRunDataCache == null)
+            {
+                return;
+            }
+
+            IVsTextView vsTextView = VsShellUtilities.GetTextView(pFrame);
+            if (vsTextView == null)
+            {
+                return;
+            }
+
+            if (vsTextView.GetBuffer(out IVsTextLines vsTextLines) != VSConstants.S_OK)
+            {
+                return;
+            }
+
+            IComponentModel componentModel = (IComponentModel)Package.GetGlobalService(typeof(SComponentModel));
+            if (componentModel == null)
+            {
+                return;
+            }
```

Comment:
should this throw here?

Reply:
Not in this case, failure is okay.

---
In reply to: [488137585](https://github.com/microsoft/sarif-visualstudio-extension/pull/195#discussion_r488137585) [](ancestors = 488137585)